uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Import/Parser/BaseTest.php

```diff
@@ -0,0 +1,288 @@
+      'type_id' => $this->absenceType->id,
+      'contact_id' => 1,
+      'period_id' => $period->id
+    ]);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 3);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => 1]);
+
+    $contactId = 1;
+
+    $row1 = [
+      'contact_id' => $contactId,
+      'absence_id' => 1,
+      'absence_type' => $this->absenceType->title,
+      'absence_date' => '2016-01-01',
+      'qty' => 2,
+      'start_date' => '2016-01-01',
+      'end_date' => '2016-01-02',
+      'total_qty' => 3,
+      'status' => 'Approved',
+      'comments' => 'Test Comment',
+    ];
+
+    $fields = array_keys($row1);
+    $importObject = $this->getImportObject($fields);
+
+    $row2 = $row1;
+    $row2['qty'] = 1;
+    $row2['absence_date'] = '2016-01-02';
+
+    $valuesRow1 = array_values($row1);
+    $valuesRow2 = array_values($row2);
+    $response1 = $importObject->import(NULL, $valuesRow1);
+    $response2 = $importObject->import(NULL, $valuesRow2);
+
+    $this->assertEquals(CRM_Import_Parser::VALID, $response1);
+    $this->assertEquals(CRM_Import_Parser::VALID, $response2);
+
+    $leaveRequest = new LeaveRequest();
+    $leaveRequest->from_date = CRM_Utils_Date::processDate('2016-01-01');
+    $leaveRequest->to_date = CRM_Utils_Date::processDate('2016-01-02');
+    $leaveRequest->contact_id = $contactId;
+    $leaveRequest->find(true);
+
+    $this->assertNotNull($leaveRequest->id);
+    //a total balance change of -3 is expected, i.e the value in total_qty field
+    $balanceChange = LeaveBalanceChange::getTotalBalanceChangeForLeaveRequest($leaveRequest);
+    $this->assertEquals(-3, $balanceChange);
+
+    //Check that comment was properly created
+    $leaveRequestCommentService = new LeaveRequestCommentService();
+    $result = $leaveRequestCommentService->get(['leave_request_id' => $leaveRequest->id, 'sequential' => 1]);
+    $this->assertEquals($result['values'][0]['text'], $row1['comments']);
+    $this->assertEquals($result['values'][0]['contact_id'], $row1['contact_id']);
+    $commentDate = new DateTime('2016-01-01');
+    $expectedCommentDate = new DateTime($result['values'][0]['created_at']);
+    $this->assertEquals($expectedCommentDate, $commentDate);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage Contact does not have period entitlement for the absence type
+   */
+  public function testValidateLeaveParamsThrowsExceptionWhenContactHasNoPeriodEntitlementForTheAbsenceType() {
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
+    ]);
+
+    $contactId = 1;
+    $row1 = [
+      'contact_id' => $contactId,
```

Comment:
You're not using this $contactId variable to anything else other than this, so I think you can drop it and just pass 1 here